tests/e2e/tidbcluster/serial.go

```diff
@@ -1136,7 +1032,63 @@ var _ = ginkgo.Describe("[tidb-operator][Serial]", func() {
 				klog.Infof("confirm tikv pods haven't been changed this time")
 				return false, nil
 			})
-			framework.ExpectError(err, "expect tikv and pd haven't been changed for 5 minutes")
+			framework.ExpectEqual(err, wait.ErrWaitTimeout, "expect tikv and pd haven't been changed for 5 minutes")
+		})
+	})
+
+	ginkgo.Context("upgrading tidb-operator in the same minor series should not trigger rolling-update", func() {
```

Comment:
this case can be covered by unit or integration tests but that requires us to maintain some test data.  it's easier to have a simple e2e test for now.